com.hazelcast.internal.networking.nio.NioChannelMemoryLeakTest.testNioChannelLeak_afterMultipleSplitBrainMerges [HZ-1685]

_master_ (commit 68b73ea697ff72cd210b19f41479d07029455776)

Failed on Sonar build (Oracle JDK 11): https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-sonar/1260/testReport/com.hazelcast.internal.networking.nio/NioChannelMemoryLeakTest/testNioChannelLeak_afterMultipleSplitBrainMerges/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: Cluster size is not correct. Expected: 3, actual: 2, instance index: 0
	at org.junit.Assert.fail(Assert.java:89)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:1021)
	at com.hazelcast.test.HazelcastTestSupport.lambda$assertClusterSizeEventually$9(HazelcastTestSupport.java:1046)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1226)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1243)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1046)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1034)
	at com.hazelcast.internal.networking.nio.NioChannelMemoryLeakTest.testNioChannelLeak_afterMultipleSplitBrainMerges(NioChannelMemoryLeakTest.java:86)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

</details>

<details><summary>Standard output:</summary>

```
02:16:15,133  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MetricsConfigHelper] testNioChannelLeak_afterMultipleSplitBrainMerges - [LOCAL] [dev] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:16:15,134  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [logo] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:16:15,134  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
02:16:15,134  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220627 - 68b73ea) starting at [127.0.0.1]:5701
02:16:15,134  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Cluster name: dev
02:16:15,134  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
02:16:15,134  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true)
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true (for Hazelcast embedded, works only when loading config via Config.load)
  - Add environment variable: HZ_JET_ENABLED=true (recommended when running container image. For Hazelcast embedded, works only when loading config via Config.load)
02:16:15,136  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MetricsConfigHelper] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:16:15,140  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Using TCP/IP discovery
02:16:15,140  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [CPSubsystem] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:16:15,144  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Diagnostics] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:16:15,144  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTING
02:16:15,146  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpIpJoiner] hz.thirsty_shannon.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is added to the blacklist.
02:16:15,146  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpIpJoiner] hz.thirsty_shannon.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is added to the blacklist.
02:16:16,146  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 this
]

02:16:16,147  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTED
02:16:16,147  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MetricsConfigHelper] testNioChannelLeak_afterMultipleSplitBrainMerges - [LOCAL] [dev] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:16:16,148  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [logo] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:16:16,148  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
02:16:16,148  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220627 - 68b73ea) starting at [127.0.0.1]:5702
02:16:16,148  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Cluster name: dev
02:16:16,148  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
02:16:16,148  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true)
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true (for Hazelcast embedded, works only when loading config via Config.load)
  - Add environment variable: HZ_JET_ENABLED=true (recommended when running container image. For Hazelcast embedded, works only when loading config via Config.load)
02:16:16,148  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [HealthMonitor] hz.thirsty_shannon.HealthMonitor - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] processors=8, physical.memory.total=755.8G, physical.memory.free=227.6G, swap.space.total=8.0G, swap.space.free=8.0G, heap.memory.used=418.9M, heap.memory.free=358.1M, heap.memory.total=777.0M, heap.memory.max=2.0G, heap.memory.used/total=53.88%, heap.memory.used/max=20.44%, minor.gc.count=6345, minor.gc.time=81836ms, major.gc.count=4, major.gc.time=2661ms, load.process=0.00%, load.system=0.00%, load.systemAverage=19.46, thread.count=453, thread.peakCount=2344, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=0, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=0, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
02:16:16,152  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MetricsConfigHelper] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:16:16,159  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Using TCP/IP discovery
02:16:16,159  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [CPSubsystem] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:16:16,177  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Diagnostics] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:16:16,177  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is STARTING
02:16:16,179  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpIpJoiner] hz.dazzling_shannon.cached.thread-3 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is added to the blacklist.
02:16:16,184  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.thirsty_shannon.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:49213
02:16:16,184  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:49213 and /127.0.0.1:5701
02:16:16,286  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.thirsty_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 this
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa
]

02:16:16,287  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.dazzling_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa this
]

02:16:16,287  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is STARTED
02:16:16,288  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MetricsConfigHelper] testNioChannelLeak_afterMultipleSplitBrainMerges - [LOCAL] [dev] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:16:16,289  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [logo] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:16:16,289  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
02:16:16,289  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220627 - 68b73ea) starting at [127.0.0.1]:5703
02:16:16,289  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Cluster name: dev
02:16:16,289  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
02:16:16,289  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [system] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true)
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true (for Hazelcast embedded, works only when loading config via Config.load)
  - Add environment variable: HZ_JET_ENABLED=true (recommended when running container image. For Hazelcast embedded, works only when loading config via Config.load)
02:16:16,293  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MetricsConfigHelper] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:16:16,298  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Using TCP/IP discovery
02:16:16,298  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [CPSubsystem] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:16:16,307  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Diagnostics] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:16:16,307  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is STARTING
02:16:16,312  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.thirsty_shannon.IO.thread-in-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:51515
02:16:16,312  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-1 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:51515 and /127.0.0.1:5701
02:16:16,314  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.IO.thread-in-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:36649
02:16:16,314  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:36649 and /127.0.0.1:5702
02:16:16,411  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.thirsty_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 this
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa
	Member [127.0.0.1]:5703 - 70d96533-1116-48f2-aaf1-1e7b276ffaf1
]

02:16:16,411  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.xenodochial_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa
	Member [127.0.0.1]:5703 - 70d96533-1116-48f2-aaf1-1e7b276ffaf1 this
]

02:16:16,411  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.dazzling_shannon.generic-operation.thread-2 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 

Members {size:3, ver:3} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa this
	Member [127.0.0.1]:5703 - 70d96533-1116-48f2-aaf1-1e7b276ffaf1
]

02:16:16,412  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is STARTED
02:16:16,412  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=2, /127.0.0.1:5701->/127.0.0.1:51515, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=70d96533-1116-48f2-aaf1-1e7b276ffaf1, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Socket explicitly closed
02:16:16,412  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-1 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connection[id=1, /0.0.0.0:51515->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=500efe91-ce9c-4197-b5a5-df7f23adb177, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
02:16:16,412  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Removing Member [127.0.0.1]:5703 - 70d96533-1116-48f2-aaf1-1e7b276ffaf1
02:16:16,412  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 this
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa
]

02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=2, /127.0.0.1:5702->/127.0.0.1:36649, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=70d96533-1116-48f2-aaf1-1e7b276ffaf1, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Member left event received from master
02:16:16,413  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 is suspected to be dead
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connection[id=2, /0.0.0.0:36649->/127.0.0.1:5702, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=822c6405-d47d-4d51-b2c4-5959cadd49aa, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
02:16:16,413  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa is suspected to be dead
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.thirsty_shannon.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 70d96533-1116-48f2-aaf1-1e7b276ffaf1
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Starting mastership claim process...
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Local MembersView{version=3, members=[MemberInfo{address=[127.0.0.1]:5701, uuid=500efe91-ce9c-4197-b5a5-df7f23adb177, liteMember=false, memberListJoinVersion=1}, MemberInfo{address=[127.0.0.1]:5702, uuid=822c6405-d47d-4d51-b2c4-5959cadd49aa, liteMember=false, memberListJoinVersion=2}, MemberInfo{address=[127.0.0.1]:5703, uuid=70d96533-1116-48f2-aaf1-1e7b276ffaf1, liteMember=false, memberListJoinVersion=3}]} with suspected members: [Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177, Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa] and initial addresses to ask: []
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.dazzling_shannon.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa this
]

02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.dazzling_shannon.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 70d96533-1116-48f2-aaf1-1e7b276ffaf1
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.xenodochial_shannon.cached.thread-3 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5701, UUID: 500efe91-ce9c-4197-b5a5-df7f23adb177
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.xenodochial_shannon.cached.thread-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:4} [
	Member [127.0.0.1]:5703 - 70d96533-1116-48f2-aaf1-1e7b276ffaf1 this
]

02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] hz.xenodochial_shannon.cached.thread-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Mastership is claimed with: MembersView{version=4, members=[MemberInfo{address=[127.0.0.1]:5703, uuid=70d96533-1116-48f2-aaf1-1e7b276ffaf1, liteMember=false, memberListJoinVersion=3}]}
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnector] hz.xenodochial_shannon.cached.thread-5 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5701, timeout: 10000, bind-any: true
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.xenodochial_shannon.cached.thread-3 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 822c6405-d47d-4d51-b2c4-5959cadd49aa
02:16:16,413  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is MERGING
02:16:16,414  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Resetting local member UUID. Previous: 70d96533-1116-48f2-aaf1-1e7b276ffaf1, new: 59f8d30d-ab09-45c7-8800-57aa6ad5d934
02:16:16,414  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [InternalPartitionService] hz.xenodochial_shannon.migration - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Fetching partition tables from cluster to determine the most recent one... Local stamp: 0
02:16:16,414  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [InternalPartitionService] hz.xenodochial_shannon.migration - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Most recent partition table is determined.
02:16:16,414  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connection[id=3, /0.0.0.0:47375->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: TcpServer is stopping
02:16:16,414  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpIpJoiner] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is added to the blacklist.
02:16:16,416  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.thirsty_shannon.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=3, /127.0.0.1:5701->/127.0.0.1:47375, qualifier=null, endpoint=null, remoteUuid=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: Connection closed by the other side
02:16:16,432  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnector] hz.xenodochial_shannon.cached.thread-9 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5701, timeout: 10000, bind-any: true
02:16:16,433  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.thirsty_shannon.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:44013
02:16:16,433  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:44013 and /127.0.0.1:5701
02:16:16,434  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.thirsty_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:3, ver:5} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 this
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa
	Member [127.0.0.1]:5703 - 59f8d30d-ab09-45c7-8800-57aa6ad5d934
]

02:16:16,435  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.xenodochial_shannon.generic-operation.thread-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] 

Members {size:3, ver:5} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa
	Member [127.0.0.1]:5703 - 59f8d30d-ab09-45c7-8800-57aa6ad5d934 this
]

02:16:16,435  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.dazzling_shannon.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 

Members {size:3, ver:5} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa this
	Member [127.0.0.1]:5703 - 59f8d30d-ab09-45c7-8800-57aa6ad5d934
]

02:16:16,436  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnector] hz.xenodochial_shannon.cached.thread-7 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 10000, bind-any: true
02:16:16,436  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.IO.thread-in-2 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:52343
02:16:16,436  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-1 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:52343 and /127.0.0.1:5702
02:16:16,440  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5703 and /127.0.0.1:52665
02:16:16,440  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:52665 and /127.0.0.1:5703
02:16:16,615  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterJoinManager] hz.thirsty_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] New join request has been received from an existing endpoint Member [127.0.0.1]:5703 - 59f8d30d-ab09-45c7-8800-57aa6ad5d934. Removing old member and processing join request...
02:16:16,615  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] hz.thirsty_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Removing Member [127.0.0.1]:5703 - 59f8d30d-ab09-45c7-8800-57aa6ad5d934
02:16:16,615  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=4, /127.0.0.1:5702->/127.0.0.1:52343, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=59f8d30d-ab09-45c7-8800-57aa6ad5d934, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Member left event received from master
02:16:16,616  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.thirsty_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:6} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 this
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa
]

02:16:16,616  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.thirsty_shannon.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 59f8d30d-ab09-45c7-8800-57aa6ad5d934
02:16:16,616  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=3, /127.0.0.1:52665->/127.0.0.1:5703, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=59f8d30d-ab09-45c7-8800-57aa6ad5d934, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Member left event received from master
02:16:16,616  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-1 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connection[id=5, /0.0.0.0:52343->/127.0.0.1:5702, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=822c6405-d47d-4d51-b2c4-5959cadd49aa, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
02:16:16,616  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.dazzling_shannon.cached.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 59f8d30d-ab09-45c7-8800-57aa6ad5d934
02:16:16,616  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.dazzling_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:6} [
	Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177
	Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa this
]

02:16:16,617  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connection[id=6, /127.0.0.1:5703->/127.0.0.1:52665, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=822c6405-d47d-4d51-b2c4-5959cadd49aa, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
02:16:16,617  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnector] hz.xenodochial_shannon.cached.thread-6 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 10000, bind-any: true
02:16:16,617  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.IO.thread-in-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:39667
02:16:16,618  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.IO.thread-in-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:39667 and /127.0.0.1:5702
02:16:17,436  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] testNioChannelLeak_afterMultipleSplitBrainMerges - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is MERGED
02:16:21,302  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ExplicitSuspicionOp] hz.xenodochial_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Received suspicion request from: [127.0.0.1]:5701
02:16:21,302  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] hz.xenodochial_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177 is suspected to be dead for reason: explicit suspicion
02:16:21,302  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ExplicitSuspicionOp] hz.xenodochial_shannon.generic-operation.thread-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Received suspicion request from: [127.0.0.1]:5701
02:16:21,302  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.priority-generic-operation.thread-0 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connection[id=4, /127.0.0.1:44013->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=500efe91-ce9c-4197-b5a5-df7f23adb177, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: explicit suspicion
02:16:21,302  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.thirsty_shannon.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=4, /127.0.0.1:5701->/127.0.0.1:44013, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=59f8d30d-ab09-45c7-8800-57aa6ad5d934, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
02:16:21,303  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ExplicitSuspicionOp] hz.xenodochial_shannon.generic-operation.thread-3 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Received suspicion request from: [127.0.0.1]:5702
02:16:21,303  WARN |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] hz.xenodochial_shannon.generic-operation.thread-3 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa is suspected to be dead for reason: explicit suspicion
02:16:21,303  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.xenodochial_shannon.generic-operation.thread-3 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Connection[id=7, /127.0.0.1:39667->/127.0.0.1:5702, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=822c6405-d47d-4d51-b2c4-5959cadd49aa, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: explicit suspicion
02:16:21,303  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] hz.xenodochial_shannon.generic-operation.thread-3 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Starting mastership claim process...
02:16:21,303  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.dazzling_shannon.IO.thread-in-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=5, /127.0.0.1:5702->/127.0.0.1:39667, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=59f8d30d-ab09-45c7-8800-57aa6ad5d934, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
02:16:21,303  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] hz.xenodochial_shannon.generic-operation.thread-3 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Local MembersView{version=5, members=[MemberInfo{address=[127.0.0.1]:5701, uuid=500efe91-ce9c-4197-b5a5-df7f23adb177, liteMember=false, memberListJoinVersion=1}, MemberInfo{address=[127.0.0.1]:5702, uuid=822c6405-d47d-4d51-b2c4-5959cadd49aa, liteMember=false, memberListJoinVersion=2}, MemberInfo{address=[127.0.0.1]:5703, uuid=59f8d30d-ab09-45c7-8800-57aa6ad5d934, liteMember=false, memberListJoinVersion=5}]} with suspected members: [Member [127.0.0.1]:5701 - 500efe91-ce9c-4197-b5a5-df7f23adb177, Member [127.0.0.1]:5702 - 822c6405-d47d-4d51-b2c4-5959cadd49aa] and initial addresses to ask: []
02:16:21,304  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.xenodochial_shannon.cached.thread-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5701, UUID: 500efe91-ce9c-4197-b5a5-df7f23adb177
02:16:21,304  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [ClusterService] hz.xenodochial_shannon.cached.thread-4 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:6} [
	Member [127.0.0.1]:5703 - 59f8d30d-ab09-45c7-8800-57aa6ad5d934 this
]

02:16:21,304  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [MembershipManager] hz.xenodochial_shannon.cached.thread-4 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Mastership is claimed with: MembersView{version=6, members=[MemberInfo{address=[127.0.0.1]:5703, uuid=59f8d30d-ab09-45c7-8800-57aa6ad5d934, liteMember=false, memberListJoinVersion=5}]}
02:16:21,304  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TransactionManagerService] hz.xenodochial_shannon.cached.thread-2 - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 822c6405-d47d-4d51-b2c4-5959cadd49aa
02:16:21,554  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [InternalPartitionService] hz.xenodochial_shannon.migration - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Fetching partition tables from cluster to determine the most recent one... Local stamp: 0
02:16:21,554  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [InternalPartitionService] hz.xenodochial_shannon.migration - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Most recent partition table is determined.
02:18:17,774  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] main - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
02:18:17,775  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Shutting down connection manager...
02:18:17,776  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] main - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=1, /0.0.0.0:49213->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=500efe91-ce9c-4197-b5a5-df7f23adb177, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: TcpServer is stopping
02:18:17,776  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnection] hz.thirsty_shannon.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=1, /127.0.0.1:5701->/127.0.0.1:49213, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=822c6405-d47d-4d51-b2c4-5959cadd49aa, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
02:18:17,777  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Shutting down node engine...
02:18:17,777  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnector] hz.thirsty_shannon.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 10000, bind-any: true
02:18:17,778  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [TcpServerConnector] hz.thirsty_shannon.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Could not connect to: /127.0.0.1:5702. Reason: IOException[Connection refused to address /127.0.0.1:5702]
02:18:17,781  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [NodeExtension] main - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Destroying node NodeExtension.
02:18:17,781  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 7 ms.
02:18:17,781  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] main - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
02:18:17,781  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
02:18:17,781  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down connection manager...
02:18:17,783  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down node engine...
02:18:17,784  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [NodeExtension] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Destroying node NodeExtension.
02:18:17,784  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 3 ms.
02:18:17,784  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
02:18:17,784  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] main - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is SHUTTING_DOWN
02:18:17,785  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Shutting down connection manager...
02:18:17,793  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Shutting down node engine...
02:18:17,796  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [NodeExtension] main - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Destroying node NodeExtension.
02:18:17,796  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [Node] main - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 11 ms.
02:18:17,796  INFO |testNioChannelLeak_afterMultipleSplitBrainMerges| - [LifecycleService] main - [127.0.0.1]:5703 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5703 is SHUTDOWN
BuildInfo right after testNioChannelLeak_afterMultipleSplitBrainMerges(com.hazelcast.internal.networking.nio.NioChannelMemoryLeakTest): BuildInfo{version='5.2-SNAPSHOT', build='20220627', buildNumber=20220627, revision=68b73ea, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testNioChannelLeak_afterMultipleSplitBrainMerges(com.hazelcast.internal.networking.nio.NioChannelMemoryLeakTest):'
02:16:15, accumulated pauses: 70 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:16:20, accumulated pauses: 58 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:16:25, accumulated pauses: 66 ms, max pause: 8 ms, pauses over 1000 ms: 0
02:16:30, accumulated pauses: 71 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:16:35, accumulated pauses: 52 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:16:40, accumulated pauses: 48 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:16:45, accumulated pauses: 64 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:16:50, accumulated pauses: 51 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:16:55, accumulated pauses: 58 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:17:00, accumulated pauses: 70 ms, max pause: 4 ms, pauses over 1000 ms: 0
02:17:05, accumulated pauses: 51 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:17:10, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:17:15, accumulated pauses: 51 ms, max pause: 1 ms, pauses over 1000 ms: 0
02:17:20, accumulated pauses: 58 ms, max pause: 6 ms, pauses over 1000 ms: 0
02:17:25, accumulated pauses: 68 ms, max pause: 5 ms, pauses over 1000 ms: 0
02:17:30, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:17:35, accumulated pauses: 46 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:17:40, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:17:45, accumulated pauses: 119 ms, max pause: 7 ms, pauses over 1000 ms: 0
02:17:50, accumulated pauses: 60 ms, max pause: 5 ms, pauses over 1000 ms: 0
02:17:55, accumulated pauses: 45 ms, max pause: 2 ms, pauses over 1000 ms: 0
02:18:00, accumulated pauses: 107 ms, max pause: 5 ms, pauses over 1000 ms: 0
02:18:05, accumulated pauses: 81 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:18:10, accumulated pauses: 63 ms, max pause: 3 ms, pauses over 1000 ms: 0
02:18:15, accumulated pauses: 271 ms, max pause: 229 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/internal/networking/nio/NioChannelMemoryLeakTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/networking/nio/NioChannelMemoryLeakTest.java
@@ -77,6 +77,8 @@ public class NioChannelMemoryLeakTest extends HazelcastTestSupport {
         assertClusterSizeEventually(3, instance1, instance2, instance3);
 
         for (int i = 0; i < 5; i++) {
+            sleepSeconds(1);
+
             closeConnectionBetween(instance1, instance3);
             closeConnectionBetween(instance2, instance3);
             assertClusterSizeEventually(2, instance1, instance2);

